ext/posix/posix.c

```diff
@@ -1208,8 +1208,12 @@ PHP_FUNCTION(posix_pathconf)
 		Z_PARAM_LONG(name);
 	ZEND_PARSE_PARAMETERS_END();
 
-	if (path_len == 0 || php_check_open_basedir(path)) {
-		RETURN_FALSE;
+	if (path_len == 0) {
+		zend_value_error("Empty path supplied");
+		RETURN_THROWS();
+	} else if (php_check_open_basedir(path)) {
+		zend_value_error("Invalid path supplied: %s", path);
+		RETURN_THROWS();
 	}
```

Comment:
Hmm, throwing a `ValueError` on an `open_basedir` violation might not be the best idea. After all, it might be hard to check in advance. Are we doing that elsewhere (instead of just failing with `E_WARNING`).

Reply:
> Are we doing that elsewhere (instead of just failing with E_WARNING)

Would return false and raising E_WARNING in both cases would be good enough.

Reply:
> Would return false and raising E_WARNING in both cases would be good enough ?

Yeah, that would be good enough, but a `ValueError` on an empty string, and `E_WARNING` on `open_basedir` violation would even be better, in my opinion. My point is that it is trivial to avoid passing an empty string (and it is clearly a programmer error to pass an empty string), but it's hard to check for `open_basedir` issues in advance.